### ISSUE TITLE
Weird indentation change in 3.9.x from `braces` rule

### ISSUE THREAD

**Issue by zoispag** [FriendsOfPHP/PHP-CS-Fixer#6519]
## Bug report

There's a weird indentation change suggested by the `braces` rule in the 3.9.x version of PHP CS Fixer with an anonymous function as a parameter to a methid, inside a `match` expression.

Version:

```
PHP CS Fixer 3.9.5 Grand Awaiting by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.1.8
```

Command:

```bash
php-cs-fixer fix --dry-run --diff --show-progress=dots --config=.php-cs-fixer.php
```

Config file:

```php
$rules = [
    '@PSR12' => true,
    '@Symfony' => true,

    'binary_operator_spaces' => ['default' => 'single_space', 'operators' => ['=>' => null]],
    'blank_line_before_statement' => false,
    'blank_line_between_import_groups' => false,
    'concat_space' => ['spacing' => 'one'],
    'echo_tag_syntax' => ['format' => 'short'],
    'increment_style' => ['style' => 'post'],
    'multiline_whitespace_before_semicolons' => ['strategy' => 'no_multi_line'],
    'no_alias_language_construct_call' => true,
    'no_alias_functions' => true,
    'no_superfluous_phpdoc_tags' => false,
    'no_unreachable_default_argument_value' => true,
    'no_useless_return' => true,
    'not_operator_with_successor_space' => true,
    'phpdoc_summary' => false,
    'php_unit_method_casing' => ['case' => 'snake_case'],
    'protected_to_private' => false,
    'psr_autoloading' => true,
    'self_accessor' => true,
    'semicolon_after_instruction' => false,
    'simplified_null_return' => false,
    'switch_continue_to_break' => false,
    'yoda_style' => false,
];
```

### Code snippet that reproduces the problem

```php
return match ($outputType) {
    self::OUTPUT_TYPE_PHP => $query->lazy(10000)
        ->map(function ($row) use ($tz) {
            $row->moment = Carbon::createFromTimestamp($row->moment, $tz);

            return $row;
        }),
};
```

PHP CS Fixer suggests the following change which looks weird:

```diff
return match ($outputType) {
    self::OUTPUT_TYPE_PHP => $query->lazy(10000)
        ->map(function ($row) use ($tz) {
            $row->moment = Carbon::createFromTimestamp($row->moment, $tz);

            return $row;
-        }),
+        }
+        ),
};
```

I could accept: 

```diff
return match ($outputType) {
    self::OUTPUT_TYPE_PHP => $query->lazy(10000)
        ->map(function ($row) use ($tz) {
            $row->moment = Carbon::createFromTimestamp($row->moment, $tz);

            return $row;
-        }),
+        }
+    ),
};
```

but having `}` and `)` in different lines with the same indentation looks weird. 

Interestingly, the same indentation works fine if I use a lookup table:

```php
return [
    self::OUTPUT_TYPE_PHP => $query->lazy(10000)
        ->map(function ($row) use ($tz) {
            $row->moment = Carbon::createFromTimestamp($row->moment, $tz);

            return $row;
        }),
][$outputType];
```

**Comment by julienfalque** [comment]
This should be fixed by #4884.

**Comment by Seldaek** [comment]
I also see a similar issue here whereas the `braces` rule indents a comment here in a nonsensical way (to me anyway..)

```diff
<?php

$key = '';
switch ($key) {
	// lala
	case 'foo':
		// lala
		$a = '';

		return $a;

-	// lulu
+		// lulu
	case 'bar':
		// lala
		$a = '';

		return $a;
}
```

@julienfalque any clue if this is also fixed by your PR?

**Comment by julienfalque** [comment]
Reported issue was fixed by #4884.

@Seldaek this might be fixed by #6490 depending on its outcome.
